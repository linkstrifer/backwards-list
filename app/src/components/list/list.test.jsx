import React from 'react';
import renderer from 'react-test-renderer';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ListComponent from './list.component';


Enzyme.configure({ adapter: new Adapter() });

const testList = `<ul>
  <li>item 1</li>
  <li>item 2</li>
  <li>item 3</li>
  <li>item 4</li>
  <li>item 9</li>
  <li>item 10</li>
  <li>item 5</li>
  <li>item 7</li>
  <li>item 8</li>
  <li>item 6</li>
  </ul>
`;

test('List snapshot', () => {
  const component = renderer.create(<ListComponent list={testList} />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
