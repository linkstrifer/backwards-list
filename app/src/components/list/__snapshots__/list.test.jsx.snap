// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-around",
      }
    }
  >
    <input
      onInput={[Function]}
      placeholder="buscar"
    />
    <div>
      <label
        htmlFor="input-order"
        style={
          Object {
            "marginRight": "10px",
          }
        }
      >
        Ordenar:
      </label>
      <select
        id="input-order"
        onChange={[Function]}
      >
        <option
          defaultValue={true}
          value="asc"
        >
          a...z
        </option>
        <option
          value="des"
        >
          z...a
        </option>
      </select>
    </div>
  </div>
  <ul>
    <li>
      item 1
    </li>
    <li>
      item 10
    </li>
    <li>
      item 2
    </li>
    <li>
      item 3
    </li>
    <li>
      item 4
    </li>
    <li>
      item 5
    </li>
    <li>
      item 6
    </li>
    <li>
      item 7
    </li>
    <li>
      item 8
    </li>
    <li>
      item 9
    </li>
  </ul>
</div>
`;
